docs/tutorials/vqe_with_estimator.ipynb

```diff
@@ -2,335 +2,447 @@
  "cells": [
```

Comment:
Add the footer.

Keeping track of the cost function eval per iteration is a bit tricky.  Namely, I need to pass the `Estimator` to the callback to do the evaluation.  However, using a `Session` in the context means that doing so would require the entire callback to be defined in the context.  Of course, one could still define sessions outside of the context and then defining the callback after that, but it gets a bit messy.  However, seeing the convergence is more or less a standard thing, so I am happy to refactor at the cost of simplicity if that is what people want.

Reply:
So yeah, the solution would be to write a function that takes the estimator to be used, and returns a callback that utilizes the estimator to do the energy evaluation.  I had to do something similar to get get `args` passed to the SPSA in Qiskit that does not allow args; another reason to move it to SciPy.

Honestly it is a minimal amount of work, and who does not like more plots?.  So let me refactor.

Reply:
Also, one of the things to keep in mind here is that the cost function itself need not be evaluated at the point of interest.  For example, SPSA does not evaluate the function at the desired vector, but rather at random offsets from this to compute the gradient.  Therefore, if not careful, one is not seeing what they think they are.

Reply:
I have made the changes, but the runtime is currently taking ~60sec per iteration.  So it will take a couple of hours for this to get into final state.